### ISSUE TITLE
Stratified QQ

### ISSUE THREAD

**Issue by swvanderlaan** [YinLiLin/R-CMplot#13]
Hi,

It would be great if you could add in a function to make stratified QQ plots. For instance stratified by bins of info-score (_e.g._ https://github.com/swvanderlaan/MetaGWASToolKit/blob/master/SCRIPTS/plotter.qq_by_info.R) and minor allele frequency (_e.g._ https://github.com/swvanderlaan/MetaGWASToolKit/blob/master/SCRIPTS/plotter.qq_by_caf.R). These are great diagnostic tools to review which the best filtering settings are for the data.

Best,

Sander

**Comment by YinLiLin** [comment]
Hi Sander,

Thank you for your suggestion, i saw your well-written script, it is a good reference to follow up, I will have a try and achieve it in CMplot with your permission.

regards,
Lilin

**Comment by swvanderlaan** [comment]
Thanks for the compliment 👨🏻‍💻😁

That would be great. Please go ahead...

**Comment by swvanderlaan** [comment]
Any progress on this?

**Comment by YinLiLin** [comment]
Oh, very sorry for that.
I missed your response here, apologise for it. I remembered that I checked your script, it seems that we need MAF or other Information to achieve it? am I right? if yes, it maybe a little hard to incorporate this function with CMplot, as CMplot only requires ```SNP Che Pos Pvalue1 Pvalue2 Pvalue3...```, which are generally provided by lots of GWAS soft wares and can be easily prepared by users.

**Comment by swvanderlaan** [comment]
I would argue that `MAF` and `INFO` are available for each GWAS. My idea would be to have stratified QQ plots including lambda's per bin (and potentially counts of variants), as a way to assess the raw results from GWAS prior to filtering them. Below an example. 

![image](https://user-images.githubusercontent.com/8877879/81670020-518ceb00-9447-11ea-8357-3be944577033.png)

These plots are quite informative. 

You are right, that for final GWAS summary statistics from _meta-analyses_, `INFO` might not be available, but for every GWAS the `MAF` or `CAF` or `EAF` or `AF` should be available. And all GWAS softwares that I work with - SNPTEST and PLINK for instance -  produce (raw) results with these variables. 

It would be another type of course and yes, one would have to supply these data as a pre-requisite, `plot_type = "qs"` for instance. 

Would be great.

**Comment by YinLiLin** [comment]
Thanks for providing the examples, I agree with you, and the stratified QQplot using MAF information is definitely worth to have a try. To avoid breaking the structure of current data format, how about adding a parameter 'maf' in CMplot allowing users to input the MAF information? then we can use it to drew the figure1 you shown above.

**Comment by swvanderlaan** [comment]
Yes that would be a great idea.

**Comment by YinLiLin** [comment]
I found it can be achieved on the current version of CMplot when I was in process of tweaking the script, an example was shown below:
```r
library(CMplot)
data(pig60K)

# maf generating
set.seed(123)
maf=0.001+0.45 * runif(nrow(pig60K))

# group assigning on basis of maf
p1=p2=p3=rep(NA, nrow(pig60K))
p1[maf<0.05]=pig60K$trait1[maf<0.05]
p2[maf<0.1&maf>=0.05]=pig60K$trait1[maf<0.1&maf>=0.05]
p3[maf>0.1]=pig60K$trait1[maf>0.1]
data=cbind(pig60K[,1:3], pig60K$trait1, p1, p2, p3)
colnames(data)[-c(1:3)]=c("All", "maf<0.05", "0.05=<maf<0.1","maf>=0.1")

# plot
CMplot(data, plot.type="q", multracks=T, conf.int=F)
```
The final visualised result:
<img width="653" alt="image" src="https://user-images.githubusercontent.com/12557293/85825812-8de19400-b7c6-11ea-8904-1fc9240b6dff.png">

Is it consistent with the figures you mentioned above? 
That means we need to adjust the format of the data manually prior to plotting.

**Comment by swvanderlaan** [comment]
Yes, this is a good start. Very good. Only thing lacking is the lambda per bin. That way you can assess whether there is inflation originating from that bin or not.
